381 update import script to json payload

Elkrival opened the pull request:
This pr updates the script to parse csv files and makes a requests to the api. It no longer depends on pymongo to parse the database and make the updates.

The pr removes unused code, and it adds a service that collects the parsed data, when the data is collected it then makes a request to the backend node api to insert the data.

It also adds a configuration of api_url which points to the server route.

Tests and more tests.

Figma link to workflow. https://www.figma.com/file/EYWZPBDHD25Of0KUkwMksC/import-Workflow?type=whiteboard&node-id=0-1&t=DDCkCP8SeU9rSRy6-0

elmantis committed:
u
Load Test CSV

* This pr adds a script generate_test_csv to generate import files that stress test the api endpoint
* It uses parallelism to write the files and it is configurable using the yaml file

 # json

* Added hash collection

elmantis committed:
Service

* Added service that parses data csv and metadata csv
* Using the new data structures the service parses the data and creats a
  json
* Added tests to the service

Updated import script for v2

* Removed all files that parse mongodb and removed dependency
* Added service that handles incoming csv data for metadata and
  participant csv data
* Service also has a method to convert data to json
* Added api request with a configuration for url

Add requests

* Added requests dependency

Updates to script to add hash collection

* Added hash collection

elmantis committed:
Service

* Converted api functions to import service
* Added api auth keys to service

 # participant csv data

 # json

 # participant csv data

ngmaloney commented on scripts/api/api.py:
```diff
@@ -0,0 +1,23 @@
+        headers=headers,
+        data=data_file_data,
+    )
+    print(r)
```
I don't think it should print. Maybe a `log.info`?

Elkrival replied:
https://github.com/AMP-SCZ/dpimport/commit/17a12ba14df570de31c11435e4ade8b837bbb3a0

ngmaloney commented on scripts/api/api.py:
```diff
@@ -0,0 +1,23 @@
+        headers=headers,
+        data=metadata_file_data,
+    )
+    print(r)
```
Same as above. Now that I think about it, we probably want to return the status or success of r?

Elkrival replied:
https://github.com/AMP-SCZ/dpimport/commit/17a12ba14df570de31c11435e4ade8b837bbb3a0

taylorkearns commented on scripts/data_importer_service/data_import_service.py:
```diff
@@ -0,0 +1,131 @@
+
+    def diverge_files(self, path):
+        basename = os.path.basename(path)
+        is_data_file = self.DATAFILE.match(basename)
```
What do you think about calling these vars `data_file` and `metadata_file`? It's not a boolean expression, I believe it returns the matched regex or `None`.

Elkrival replied:
[733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573)

taylorkearns commented on scripts/data_importer_service/data_import_service.py:
```diff
@@ -0,0 +1,131 @@
+        is_metadata_file = self.METADATA.match(basename)
+
+        if is_data_file:
+            file_extension_to_dict = self.DATAFILE.match(basename).groupdict()
```
Can use the var declared on 26 here.

Elkrival replied:
[733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573)

taylorkearns commented on scripts/data_importer_service/data_import_service.py:
```diff
@@ -0,0 +1,131 @@
+        is_metadata_file = self.METADATA.match(basename)
+
+        if is_data_file:
+            file_extension_to_dict = self.DATAFILE.match(basename).groupdict()
```
```suggestion
            data_file_matches = self.DATAFILE.match(basename).groupdict()
```

Elkrival replied:
[733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573)

taylorkearns commented on scripts/data_importer_service/data_import_service.py:
```diff
@@ -0,0 +1,131 @@
+
+            return self.process_data_file(path, file_extension_to_dict)
+        if is_metadata_file:
+            metadata_file_extension = self.METADATA.match(basename).groupdict()
```
```suggestion
            metadata_file_matches = self.METADATA.match(basename).groupdict()
```

Elkrival replied:
[733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573)

taylorkearns commented on scripts/data_importer_service/data_import_service.py:
```diff
@@ -0,0 +1,138 @@
+        self.data_file_list = data_file_list
+        self.metadata_file_list = metadata_file_list
+
+    def diverge_files(self, path):
```
```suggestion
    def process_file(self, path):
```

Elkrival replied:
[733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573)

ngmaloney commented:
@Elkrival Is there an example config containing the new URL directive pointing to the API endpoint? I couldn't find it.

Elkrival commented:
@ngmaloney [733d948](https://github.com/AMP-SCZ/dpimport/pull/17/commits/733d9485a4b48da9dc28c753b9788357e5c82573) this commit updates the example config

jyurek commented on README.md:
```diff
@@ -1,40 +1,40 @@
+
 DPimport is a command line tool for importing files into DPdash using a
-simple [`glob`](https://en.wikipedia.org/wiki/Glob_(programming)) expression.
+simple [`glob`](<https://en.wikipedia.org/wiki/Glob_(programming)>) expression.
```
Why does this need the `<>`s? Is it because of the parens in the url?

Elkrival replied:
There were pushes done to main so maybe this was done for a reason. I'm unsure what.

jyurek commented on README.md:
```diff
@@ -55,18 +55,9 @@ and so on.
-
-
-
-## MongoDB
```
Is it worth noting something like "This used to require Mongo but doesn't anymore"?

Elkrival replied:
The app still uses mongo, but we don't write to it directly anymore

zelaznik commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+logger = logging.getLogger(__name__)
+
+def main():
+    parser = ap.ArgumentParser()
+    parser.add_argument('-c', '--config')
+    parser.add_argument('-d', '--dbname', default='dpdata')
+    parser.add_argument('-v', '--verbose', action='store_true')
+    parser.add_argument('expr')
+    args = parser.parse_args()
```
👍 I'm a fan of the argparse library.

Elkrival replied:
👍

zelaznik commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+    # logger.info('cleaning metadata')
+    # lastday = get_lastday(db.db)
+    # if lastday:
+    #     clean_metadata(db.db, lastday)
```
```suggestion
```

Elkrival replied:
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

jyurek commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+    logging.basicConfig(level=level)
+
+    with open(os.path.expanduser(args.config), 'r') as fo:
+        config = yaml.load(fo, Loader=yaml.SafeLoader)
```
This `config` isn't used anywhere, is it?

Elkrival replied:
yes it's used in the import.py file within the scripts directory

jyurek commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+
+    for subject in max_days:
+        if subject['_id']['study'] not in studies:
+            studies[subject['_id']['study']] = {}
+            studies[subject['_id']['study']]['subject'] = []
+            studies[subject['_id']['study']]['max_day'] = 0
```
`subject['_id']['study']` is referenced a lot in this loop. Feels like it would be worth it to both give it a clear name/reason and also to reduce noise on any given line.

Elkrival replied:
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

jyurek commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+                {
+                    '_id' : True,
+                    'collection' : True,
+                    'synced' : True
```
Does this line request that the sync status be part of the result set?

Elkrival replied:
not sure, but this code is not part of the work 
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

jyurek commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+
+            if len(metadata) > 1:
+                for doc in metadata:
+                    if doc['synced'] is False and 'collection' in doc:
+                        db[doc['collection']].drop()
```
Why does it look like you're dropping the whole collection?

Elkrival replied:
it's not part of the original work, it's been removed [655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

jyurek commented on build/scripts-3.11/import.py:
```diff
@@ -0,0 +1,191 @@
+        subject_metadata['days'] = subject['days']
+        subject_metadata['study'] = subject['_id']['study']
+
+        studies[subject['_id']['study']]['max_day'] = studies[subject['_id']['study']]['max_day'] if (studies[subject['_id']['study']]['max_day'] >= subject['days'] ) else subject['days']
```
This feels like too much for one line

Elkrival replied:
this is outdated it's been removed
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

jyurek commented on scripts/api/api.py:
```diff
@@ -0,0 +1,36 @@
+import logging
+
+
+def create_data_file(api_url, data_file_data):
+    request_url = api_url + "day"
+    headers = {"content-type": "application/json"}
+    r = requests.post(
+        request_url,
+        headers=headers,
+        data=data_file_data,
+    )
+    status = r.status_code
+    if status != 200:
+        response = r.json()["message"]
+        logging.info(response)
+    else:
+        response = r.json()["data"]
+        logging.info(response)
+
+
+def create_metadata_file(api_url, metadata_file_data):
+    request_url = api_url + "metadata"
+    headers = {"content-type": "application/json"}
+    r = requests.post(
+        request_url,
+        headers=headers,
+        data=metadata_file_data,
+    )
+    status = r.status_code
+    if status != 200:
+        response = r.json()["message"]
+        logging.info(response)
+    else:
+        response = r.json()["data"]
+        logging.info(response)
```
Unless I'm missing something, these look identical aside from the url's postfix. Could you move this functionality into a common function that both `create_data_file` and `create_metadata_file` both call?

Elkrival replied:
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)

elmantis committed:
Merge pull request #23 from AMP-SCZ/21-utility-load-test-csv

Test CSV Load generator

elmantis committed:
Merge pull request #24 from AMP-SCZ/add-api-auth-to-requests

Add authentication to request

elmantis committed:
Change keys for columns

* Subject ID and uppercase Study keys needed to be updated for import

elmantis committed:
Key change

* updated key in metadata
* Updated tests

elmantis committed:
Handle Values

* Handle infinity and nan values
* If compute error, import as string values

elmantis committed:
String

* String values

elmantis committed:
Nan

* Fix nan

elmantis committed:
Updates

* Updates to import script
* Add assessment, study and subject properties to data
* Update Tests
* Added route to restarte metadata collection

elmantis committed:
Add updates

* Updates to import

elmantis committed:
Support for new data structure

* Updates for new data migration structure

elmantis committed:
Updated tests

* Updated tests

elmantis committed:
Test constn

* Test consent and synced

elmantis committed:
Extract variables

* extract variables from assessment

mikestone14 commented:
@Elkrival what's the status of this PR? Does it need to be rebased/merged or closed?

elmantis committed:
remove print log

* removed print

elmantis committed:
variables dictionary

* added variables as dictionaries

elmantis committed:
PR comments

* Pr comments

elmantis committed:
Merge pull request #28 from AMP-SCZ/add-variables-property

Extract variables

elmantis committed:
print struct

* print structure

ngmaloney committed:
Allow SSL verification to be bypassed

When connecting with local/self-signed/dev environments it is sometimes necessary to bypass SSL verification.
This commit adds a config option to do so but it is not the default nor is it considered secure/best-practice.

tashrifbillah committed:
omit printing of imported data

tashrifbillah committed:
omit unhelpful ssl_verify=False warning
This should also speed up the import API.

tashrifbillah committed:
Merge pull request #31 from AMP-SCZ/patch-1

Omit verbosity
